_test/test_dnd_class/test_smooth.m

```diff
@@ -0,0 +1,180 @@
+            fullpath = char(test_file.getCanonicalPath());
+        end
+
+        %% SMOOTH
+        function test_smooth_returns_dnd_object(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth(10, 'hat');
+            assertTrue(isa(d, 'd2d'));
+        end
+
+        function test_smooth_no_args(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth();
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_scalar_width_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth(100, 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_array_width_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth([100, 25], 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_resolution_shape_arg(obj)
+            skipTest('No valid agruments possible for resolution call #628');
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth([100, 201, 301], 'resolution');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_hat_shape_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth(100, 'hat');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_gaussian_shape_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            d = s.smooth(100, 'gaussian');
+
+            % TODO: assert data matches expected...
+            %assertEqualToTol(d.data.s, expected.data.s, 1e-8);
+            %assertEqualToTol(d.data.e, expected.data.e, 1e-8);
+        end
+
+        function test_smooth_raises_error_with_invalid_shape_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            assertExceptionThrown(@()s.smooth(100, 'not-shape'), 'HORACE:smooth:invalid_arguments');
+        end
+
+        function test_smooth_raises_error_with_incorrect_dimension_width_arg(obj)
+            s = d2d(obj.test_sqw_2d_fullpath);
+
+            assertExceptionThrown(@()s.smooth([10,10,10,10,10]), 'HORACE:smooth:invalid_arguments');
```

Comment:
invalid arguments->incorrect length for width argument (I would have written incorrect width length but that looks wrong. I was reminded of the time that the Labour politician Michael Foot was named defence secretary - leading to the proposed headline `Foot heads arms body`.)